Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider

```diff
@@ -166,3 +162,389 @@ AS
+   AND name = N'SettingValue';
+   
+IF @colSize != -1
+    ALTER Table {databaseOwner}[{objectQualifier}HostSettings] 
+        ALTER Column SettingValue  nVarChar(Max) Not Null;
+
+-- disallow null values, as DNN cannot distinguish between empty string and null anyways.
+UPDATE {databaseOwner}[{objectQualifier}PortalSettings] 
+ SET   SettingValue = N''
+ WHERE SettingValue Is Null;
+
+SET @colSize = 0;
+SELECT @colSize = max_length 
+ FROM  sys.columns 
+ WHERE object_id = object_id('{databaseOwner}[{objectQualifier}PortalSettings]', 'U')
+   AND name = N'SettingValue';
+   
+IF @colSize != -1
+    ALTER Table {databaseOwner}[{objectQualifier}PortalSettings] 
+        ALTER Column SettingValue  nVarChar(Max) Not Null;
+  
+SET @colSize = 0;
+SELECT @colSize = max_length 
+ FROM  sys.columns 
+ WHERE object_id = object_id('{databaseOwner}[{objectQualifier}RoleSettings]', 'U')
+   AND name = N'SettingValue';
+
+   
+IF @colSize != -1
+    ALTER Table {databaseOwner}[{objectQualifier}RoleSettings] 
+        ALTER Column SettingValue nVarChar(Max) Not Null;
+GO  
+
+-- moved out of GetPortalSettings -  this doesn't need to be processed more than once:
+IF NOT EXISTS (SELECT 1 FROM {databaseOwner}[{objectQualifier}HostSettings] WHERE SettingName = 'GUID' )
+    INSERT INTO {databaseOwner}[{objectQualifier}HostSettings]
+            (SettingName, SettingValue, SettingIsSecure) 
+     VALUES ( N'GUID',    NewId(),      0              ); 
+GO
+
+-- Adjust stored procedures:
+IF EXISTS (SELECT * FROM sys.Procedures WHERE object_id = object_id(N'{databaseOwner}[{objectQualifier}GetHostSettings]'))
+    DROP PROCEDURE {databaseOwner}[{objectQualifier}GetHostSettings]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}GetHostSettings]
+AS
+BEGIN
+	SELECT SettingName,
+		   SettingValue,
+		   SettingIsSecure,
+		   CreatedByUserID,
+		   CreatedOnDate,
+	       LastModifiedByUserID,
+		   LastModifiedOnDate
+	 FROM  {databaseOwner}[{objectQualifier}HostSettings];
+END; -- Procedure
+GO
+
+
+IF EXISTS (SELECT * FROM sys.Procedures WHERE object_id = object_id(N'{databaseOwner}[{objectQualifier}GetHostSetting]'))
+    DROP PROCEDURE {databaseOwner}[{objectQualifier}GetHostSetting]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}GetHostSetting]
+    @SettingName nvarchar(50)
+AS
+BEGIN
+    SELECT SettingValue
+     FROM  {databaseOwner}[{objectQualifier}HostSettings]
+     WHERE SettingName = @SettingName;
+END; -- Procedure
+GO
+
+
+IF EXISTS (select * FROM sys.Procedures WHERE object_id = object_id(N'{databaseOwner}[{objectQualifier}UpdateHostSetting]'))
+	DROP PROCEDURE {databaseOwner}[{objectQualifier}UpdateHostSetting]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}UpdateHostSetting]
+	@SettingName	   nvarchar( 50),
+	@SettingValue	   nvarchar(Max),
+	@SettingIsSecure   bit,
+	@UserID	           int
+AS
+BEGIN
+	UPDATE {databaseOwner}[{objectQualifier}HostSettings]
```

Comment:
you are right, I used the update to update the correct column and forgot to remove the statement